emails/tests/views_tests.py

```diff
@@ -1270,36 +1315,406 @@ def test_complaint_disables_mask(self):
+        ]
+        complaint_body = {"Message": json.dumps(simulator_complaint_message)}
+
+        with (
+            self.assertLogs(INFO_LOG) as logs,
+            self.assertLogs(ERROR_LOG) as error_logs,
+        ):
+            response = _sns_notification(complaint_body)
+        assert response.status_code == 404
+
+        self.user.profile.refresh_from_db()
+        assert self.user.profile.auto_block_spam is True
+        self.mock_ses_client.send_raw_email.assert_not_called()
+
+        (rec1, rec2) = logs.records
+        # Developer mode still active due to mask match
+        assert rec1.msg == "_handle_complaint: developer_mode"
+        assert getattr(rec1, "mask_id") == self.ra.metrics_id
+        assert getattr(rec1, "dev_action") == "log"
+        assert getattr(rec1, "parts") == 1
+        assert getattr(rec1, "part") == 0
+        notification_gza85 = getattr(rec1, "notification_gza85")
+        log_complaint = decode_dict_gza85(notification_gza85)
+        assert log_complaint == simulator_complaint_message
+
+        assert rec2.msg == "complaint_notification"
+
+        (err_log,) = error_logs.records
+        assert err_log.msg == "_gather_complainers: unknown complainedRecipient"
+
+    @override_flag("developer_mode", active=True)
+    def test_complaint_developer_mode_domain_address(self):
+        """If the simulator email can not be turned into a mask, it is not changed."""
+        premium_user = make_premium_test_user()
+        assert premium_user.email
+        premium_user.profile.subdomain = "subdomain"
+        premium_user.profile.save()
+
+        domain_address = DomainAddress.objects.create(
+            address="complainer",
+            user=premium_user,
+            description="DEV:simulate_complaint",
+        )
+
+        complaint_msg = deepcopy(self.complaint_msg)
+        complaint_msg["complaint"]["complainedRecipients"] = [
+            {"emailAddress": "complaint+complainer.subdomain@simulator.amazonses.com"}
+        ]
+        complaint_msg["mail"]["commonHeaders"]["from"] = [domain_address.full_address]
+        complaint_body = {"Message": json.dumps(complaint_msg)}
+
+        with self.assertLogs(INFO_LOG) as logs:
+            response = _sns_notification(complaint_body)
+        assert response.status_code == 200
+
+        premium_user.profile.refresh_from_db()
+        assert premium_user.profile.auto_block_spam is True
+        self.mock_ses_client.send_raw_email.assert_not_called()
+
+        (rec1, rec2) = logs.records
+        assert rec1.msg == "_handle_complaint: developer_mode"
+        assert getattr(rec1, "mask_id") == domain_address.metrics_id
+        assert getattr(rec1, "dev_action") == "log"
+
+        assert rec2.msg == "complaint_notification"
+
+    def test_complaint_no_complained_recipients_error_logged(self):
+        """Without complained recipients, an error is logged, but matches on From"""
```

Comment:
suggestion (non-blocking): I had to read thru the code and test fixture to understand what this test was doing.

```suggestion
        """
        Without complainedRecipients, we can still match the user on From:,
        perform the appropriate action, and log an error
        """
```